### PR TITLE
[Token]  Publish Modules From Multiple Packages 

### PR DESCRIPTION
**Pull request by areshand**
### Description
Now, the framework is re-structured to multiple packages. However, the genesis is able to load modules from one folder. 

Modify the genesis module reading to be from all package under framework folder

### Test Plan

add unit test and manual test 

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2070)

### CODE REVIEW

File: aptos-move/af-cli/src/main.rs [hunk 1]
```diff
@@ -26,7 +26,8 @@ pub enum AfCommands {
 }
 
 fn main() -> Result<()> {
-    let error_descriptions: ErrorMapping = bcs::from_bytes(cached_framework_packages::error_map())?;
+    let error_descriptions: ErrorMapping =
+        bcs::from_bytes(&cached_framework_packages::error_map()[0])?;
```
Comment (areshand) [inline aptos-move/af-cli/src/main.rs:30]: seems this package is not used. so I just use 0 index now.

File: aptos-move/vm-genesis/src/lib.rs [hunk 2]
```diff
@@ -317,23 +334,69 @@ fn create_and_initialize_validators(
     );
 }
 
-/// Publish the standard library.
-fn publish_stdlib(session: &mut SessionExt<impl MoveResolver>, stdlib: Modules) {
-    let dep_graph = stdlib.compute_dependency_graph();
+/// Publish all modules that should be available after genesis.
+fn publish_stdlib(session: &mut SessionExt<impl MoveResolver>, stdlib: Vec<CompiledModule>) {
+    let map = construct_module_map(stdlib);
+    let root_address = AccountAddress::from_hex_literal("0x1").unwrap();
```
Comment (movekevin) [inline aptos-move/vm-genesis/src/lib.rs:340]: nit: Can we call this aptos_framework_address? Let's not mention root anywhere that's not part of test/test code if possible

File: aptos-move/vm-genesis/src/lib.rs [hunk 3]
```diff
@@ -317,23 +334,69 @@ fn create_and_initialize_validators(
...
+
+    // publish core-framework
+    publish_module_bundle(session, Modules::new(framework_modules));
+    // publish non-core-framework modules
+    publish_token_modules(session, token_modules.clone());
+}
+
+/// publish modules that are not core-framework. assuming core-framework published
+/// the modules has to be sorted by topological order PropertyMap -> TokenV1 -> TokenCoinSwap
+fn publish_token_modules(
```
Comment (movekevin) [inline aptos-move/vm-genesis/src/lib.rs:354]: How about we move the token modules into a separate dir so we don't need to maintain this hardcoded list of modules to deploy to 0x2?

File: aptos-move/framework/cached-packages/src/lib.rs [hunk 4]
```diff
@@ -2,20 +2,18 @@
...
 pub mod aptos_stdlib;
 pub mod aptos_token_stdlib;
 mod generated_aptos_txn_builder;
 mod generated_token_txn_builder;
 
-static PACKAGE: Dir<'static> = include_dir!("$OUT_DIR/framework");
+static PACKAGE: Dir<'static> = include_dir!("$OUT_DIR");
 
-static MODULE_BLOBS: Lazy<Vec<Vec<u8>>> = Lazy::new(|| load_modules("build"));
+static MODULE_BLOBS: Lazy<Vec<Vec<u8>>> = Lazy::new(load_modules);
```
Comment (gregnazario) [inline aptos-move/framework/cached-packages/src/lib.rs:16]: Did these paths change upstream?  Would this break compatibility with anything that's processing it?

File: aptos-move/vm-genesis/src/lib.rs (hunk 2 above)
Comment (areshand) [reply aptos-move/vm-genesis/src/lib.rs]: updated

File: aptos-move/vm-genesis/src/lib.rs (hunk 3 above)
Comment (areshand) [reply aptos-move/vm-genesis/src/lib.rs:354]: This is unfortunately the limitation of our publish module functions. 

To publish module bundle, it requires all modules to be from same address. So I have to publish module individually. To publish module separately, I have to handle the topological order myself. The current bundle topological sort fails as the same reason above.

File: aptos-move/framework/cached-packages/src/lib.rs (hunk 4 above)
Comment (areshand) [reply aptos-move/framework/cached-packages/src/lib.rs:16]: I validate all the callsites using this cached-package and updated them. all tests passed now.

File: aptos-move/af-cli/src/main.rs [hunk 5]
```diff
@@ -26,7 +26,8 @@ pub enum AfCommands {
 }
 
 fn main() -> Result<()> {
```
Comment (wrwg) [inline aptos-move/af-cli/src/main.rs:28]: This CLI is dead so I think you can ignore it.

File: aptos-move/vm-genesis/src/lib.rs (hunk 3 above)
Comment (wrwg) [reply aptos-move/vm-genesis/src/lib.rs:354]: Really, conceptually token is not part of the stdlib and shouldn't be considered part of genesis. Once we have the new deployment model, we need to revisit this.

**Review by wrwg — APPROVED** [review]
Just want to note that this is likely not the way how we eventually can handle the token package (i.e. as part of genesis). Because tokens would be just a regular user package, it should be deployed after genesis. But we can move forward with this for now.

**Comment by davidiw** [comment]
I think you merged this into your tokenv2 PR, closing this out, but feel free to re-open.

**Pull request closed without merging**